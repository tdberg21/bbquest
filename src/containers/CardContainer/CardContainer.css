.card-container{
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  padding: 1em;
}

.pig-gif {
  margin: 8em auto;
}