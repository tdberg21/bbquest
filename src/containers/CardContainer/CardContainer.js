import React, { Component } from 'react';
import { connect } from 'react-redux';
import Card from '../../components/Card/Card';
import './CardContainer.css';

export class CardContainer extends Component {

  restaurantsToDisplay = restaurants => restaurants.map((restaurant, index) => {
    return <Card {...restaurant} key={`${index} + ${restaurant.name}`} />;
  })

  render() {
    return (
      <div className="card-container">
        {this.restaurantsToDisplay(this.props.restaurants)}
      </div>
    );
  }
}


export const mapStateToProps = state => ({
  restaurants: state.restaurants
});

export default connect(mapStateToProps, null)(CardContainer);