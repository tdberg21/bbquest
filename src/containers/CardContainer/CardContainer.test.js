import React from 'react';
import { shallow } from 'enzyme';
import { CardContainer, mapStateToProps } from './CardContainer';

describe('CARDCONTAINER TESTS', () => {
  it('should match the snapshot with restaurants', () => {
    const mockRestaurants= [{name:'PeeWees'}];
    const wrapper = shallow(<CardContainer restaurants={mockRestaurants}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot without restaurants', () => {
    const mockRestaurants = [];
    const wrapper = shallow(<CardContainer restaurants={mockRestaurants} />);

    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should return an object with the restaurants array', () => {
      const mockState = {
        restaurants: ['PeeWees']
      };
      const expected = {
        restaurants: ['PeeWees']
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });
});