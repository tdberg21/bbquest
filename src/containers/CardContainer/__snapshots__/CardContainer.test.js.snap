// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CARDCONTAINER TESTS should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    restaurants={
        Array [
            Object {
              "name": "PeeWees",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          name="PeeWees"
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0 + PeeWees",
        "nodeType": "function",
        "props": Object {
          "name": "PeeWees",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            name="PeeWees"
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0 + PeeWees",
          "nodeType": "function",
          "props": Object {
            "name": "PeeWees",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
